core/sail/lmdb/src/main/java/org/eclipse/rdf4j/sail/lmdb/TxnManager.java

```diff
@@ -0,0 +1,253 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.sail.lmdb;
+
+import static org.eclipse.rdf4j.sail.lmdb.LmdbUtil.E;
+import static org.lwjgl.system.MemoryStack.stackPush;
+import static org.lwjgl.system.MemoryUtil.NULL;
+import static org.lwjgl.util.lmdb.LMDB.MDB_RDONLY;
+import static org.lwjgl.util.lmdb.LMDB.mdb_txn_abort;
+import static org.lwjgl.util.lmdb.LMDB.mdb_txn_begin;
+import static org.lwjgl.util.lmdb.LMDB.mdb_txn_renew;
+import static org.lwjgl.util.lmdb.LMDB.mdb_txn_reset;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.locks.StampedLock;
+
+import org.eclipse.rdf4j.sail.lmdb.LmdbUtil.Transaction;
+import org.lwjgl.PointerBuffer;
+import org.lwjgl.system.MemoryStack;
+
+/**
+ * Manager for LMDB transactions.
+ */
+class TxnManager {
+
+	private final Mode mode;
+	private final Set<Txn> active = new HashSet<>();
```

Comment:
Looks like you are using a HashSet for objects that don't implement hashCode() or equals().

Reply:
Yes, I'm relying on the default implementations. The Txn objects are only equal if they are identical. 
I think that this is a common practice. Would you propose something else?

Reply:
IdentityHashMap is typically faster and that also signals to other developers that you are intentionally relying on reference equality.

Reply:
OK, I'll use an IdentityHashMap<Txn, Boolean>